### ISSUE TITLE
Feature request: have backlinks detect standard links and not just zola (leading @) links

### ISSUE THREAD

**Issue by zmre** [Keats/gutenberg#2733]
I use zola to publish my repository of notes.  I link between notes using standard markdown with relative links like `[Note Name](../otherdir/othernote)` (no ending .md and no leading @).  Backlinks only populate if you use the `@/path/to/otherdir/othernote.md` notation, but that notation breaks all my other tools.  So in neovim, for example, when there's a path, I can use `gf` to go to that file. And my LSP detects if links are good or not and can autocomplete paths to files and their titles.  All of that goes away with the non-standard link.  My LSP also detects related notes by tracking backlinks, but of course if I switch to the Zola syntax to get that info into the rendered stuff, then I lose that info in the editor side.

So long story short, is it feasible to have backlinks populate with any link to a file and not just the zola-special absolute links?  I'd really love to be able to navigate to related notes based on backlinks without having to use the special link syntax and then break the same functionality in other places...

I'm new to Zola but very happy with it so far so great work and thank you.

**Comment by Keats** [comment]
The issue with not using the syntax is that and doing `[Note Name](../otherdir/othernote)` means that now, you cannot resolve a page markdown on its own. To get the backlinks, you would need to get all the URLs in your website and resolve them for every internal link. It's possible to do but not something I would do myself. It would also need to not affect performance in any way if it's added

**Comment by zmre** [comment]
Huh, I thought all links were already checked to make sure there were no 404s?  Is that only true for links in the custom zola syntax?

**Comment by Keats** [comment]
Custom zola syntax + external links. Other links are not checked since they could refer to anything, including content not available at zola build time
